kernel/src/engine/parquet_stats_skipping.rs

```diff
@@ -0,0 +1,1267 @@
+//! An implementation of data skipping that leverages parquet stats from the file footer.
+use crate::expressions::{BinaryOperator, Expression, Scalar, UnaryOperator, VariadicOperator};
+use crate::schema::DataType;
+use parquet::schema::types::ColumnPath;
```

Comment:
From https://github.com/delta-incubator/delta-kernel-rs/pull/357#pullrequestreview-2338728716:
> It's great that you've extracted out the stuff needed from the engine so that this PR doesn't need to reference arrow. 

We do still have this one dependency on arrow-parquet [ColumnPath](https://arrow.apache.org/rust/parquet/schema/types/struct.ColumnPath.html). But we already knew we needed to define a similar struct in order to support nested column paths. Once that struct exists, we can use it instead and push the dependency back into the concrete implementation that anyway has to know about arrow-parquet.

Reply:
(also added it as a code comment)

Reply:
added https://github.com/delta-incubator/delta-kernel-rs/issues/379